### PR TITLE
Add HolyC support

### PR DESCRIPTION
**Pull request by codingdandy**
HolyC is a C-like language, regular C highlighting doesn't work (all basic types are different, x86 Assembly is first-class).

https://en.wikipedia.org/wiki/TempleOS
http://templeos.org

Sample repositories being incorrectly categorized as C or C++ or None:
- https://github.com/minexew/TempleOS
- https://github.com/bagnalla/holyc_mal
- https://github.com/minexew/Shrine
- https://github.com/Rion96/Minesweeper
- https://github.com/minexew/Lsh
- https://github.com/Rion96/GKey
- https://github.com/Rion96/TicTacToe
- https://github.com/slx7R4GDZM/Sine

## Checklist:

- [x] **I am associating a language with a new file extension.**
  - [x] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?p=100&q=extension%3Ahc+NOT+nothack&type=Code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/minexew/TempleOS/blob/archive/Adam/Gr/GrAsm.HC
      - https://github.com/minexew/TempleOS/blob/archive/Demo/OnceDemo.HC
      - https://github.com/minexew/TempleOS/blob/archive/Demo/Prompt.HC
      - https://github.com/minexew/TempleOS/blob/archive/Demo/RandDemo.HC
      - https://github.com/minexew/TempleOS/blob/archive/Demo/Spy.HC
    - Sample license(s): Public domain
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  
  - [x] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

- [ ] **I am fixing a misclassified language**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  
  - Old: https://github-lightshow.herokuapp.com/
  - New: https://github-lightshow.herokuapp.com/
  
- [ ] **I am updating a grammar submodule**
  

- [ ] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

### CODE REVIEW

**Comment by pchaigno** [comment]
It looks like there are [a few other languages](https://github.com/search?o=desc&q=extension%3Ahc+NOT+ffhgjhdjfdb&s=indexed&type=Code) using the `.hc` file extension. We'll need to refine that search query to better estimate the number of HolyC files. Any keywords common among HolyC files?

**Comment by codingdandy** [comment]
`U0`  (always upper case, = void type) is the most common keyword, used nearly on all snippets.
Other basic types can be used for heuristic: `I8`, `U8`, `I16`, `U16`, `I32`, `U32`, `I64`, `U64`, `F64`

**Comment by pchaigno** [comment]
Thanks. It looks [widespread enough](https://github.com/search?p=1&q=extension%3Ahc+U0+OR+I64+OR+U64+OR+I32+OR+U32&type=Code&utf8=%E2%9C%93). I'll run Harvester later to make sure.

Do HolyC files always have a uppercase file extension?

**Comment by codingdandy** [comment]
Actually yes, the convention is .HC, there's also .HC.Z but it might be
compressed and not human readable (I noticed users prefer to put without .Z
on the repos), but both could be marked as HolyC if possible.

Are there any fixes I can do to the PR?

I would be happy with a initial minimal support for now, for example only
associate with uppercase .HC would work well and easy for devs to catch up
with the convention if they want the detection.

**Comment by pchaigno** [comment]
> there's also .HC.Z but it might be compressed and not human readable (I noticed users prefer to put without .Z on the repos), but both could be marked as HolyC if possible.

If those files are binaries, there not much we can do :-/

> Are there any fixes I can do to the PR?

Could you check [the Travis CI build failures](https://travis-ci.org/github/linguist/jobs/434851912) and try to fix them?

**Comment by pchaigno** [comment]
Flagging as stale.

**Comment by codingdandy** [comment]
Hi @pchaigno, my apologies for the delay.

>Could you check the Travis CI build failures and try to fix them?
Yes. `other` was not an accepted license, new commit should fix it.

**Comment by codingdandy** [comment]
Now it fails because the submodule (holyc.tmbundle) is outdated, I had some fixes on `master` there.
Any tips on how to update to the latest submodule commit? of https://github.com/codingdandy/holyc.tmbundle

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -1941,6 +1941,16 @@ HiveQL:
   tm_scope: source.hql
   ace_mode: sql
   language_id: 931814087
+HolyC:
+  type: programming
+  color: "#FFF47C"
+  extensions:
+  - ".hc"
+  - ".hc.z"
```
Comment (lildude) [inline lib/linguist/languages.yml]: Is this a compressed version of `.hc`? If so we don't need this extension as the file will be binary so ignored anyway.

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -1941,6 +1941,16 @@ HiveQL:
...
+  ace_mode: c_cpp
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: You'll have to add `tm_scope: source.hc` here.

File: lib/linguist/languages.yml [hunk 3]
```diff
@@ -1941,6 +1941,16 @@ HiveQL:
   tm_scope: source.hql
   ace_mode: sql
   language_id: 931814087
+HolyC:
+  type: programming
+  color: "#FFF47C"
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: There's a conflict on that color:
```
  1) Failure:
TestColorProximity#test_color_proximity [/home/travis/build/github/linguist/test/test_color_proximity.rb:21]:
The following 2 languages have failing color thresholds. Please modify the hex color.
- JavaScript (#f1e05a) is too close to ["FFF47C", "f1e05a"]
- Rascal (#fffaa0) is too close to ["FFF47C", "fffaa0"]
```

**Comment by codingdandy** [comment]
I ran `bundle exec rake test`, seems to have solved the color conflict after latest commit.

**Comment by pchaigno** [comment]
Could you run `script/list-grammars` to fix the build failure?

**Comment by pchaigno** [comment]
Flagging as stale.

**Pull request closed without merging**